DataFormats/Provenance/interface/Hash.h

```diff
@@ -97,8 +97,7 @@ namespace edm {
       }
       //copy constructor will do compact form conversion
       if (meCF) {
-        Hash<I> temp(iOther);
-        return op(this->hash_, temp.hash_);
```

Comment:
This seems dependent upon a side effect of the copy constructor as stated in the comment. I don’t think this change does the same work.

Reply:
ok, I will revert this change

Reply:
Note that clang-tidy in LLVM 17.0.3 automatically apply this change. So if we have to keep this copy then we should explicitly disable clang-tidy check for this section of code

Reply:
Making it a const reference continues to avoid the copy constructor.

Reply:
ah ok, I will revert this change and mark this line to be ignored by clang-tidy

Reply:
ok done